test_scripts/Defects/5_0/2379_SDL_must_send_log_error_and_ignore_invalid_RPC_in_case_SDL_cuts_off_fake_parameters_and_RPC_becomes_invalid.lua

```diff
@@ -0,0 +1,78 @@
+---------------------------------------------------------------------------------------------------
+-- User story: https://github.com/SmartDeviceLink/sdl_core/issues/2379
+--
+-- Precondition:
+-- SDL must send log error and ignore invalid RPC in case SDL cuts off fake parameters and RPC becomes invalid
+-- Description:
+-- Steps to reproduce:
+-- 1) SDL cuts off fake parameters from response (request) that SDL should use internally and this response (request) is invalid by any reason
+-- Expected:
+-- 1) Log the corresponding error
+-- 2) Ignore this response (request)
+---------------------------------------------------------------------------------------------------
+-- [[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local common = require('user_modules/sequences/actions')
+
+-- [[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+
+-- [[ Local variables ]]
+local requestParamsValidData = {
+	menuID = 10,
+	position = 5,
+	menuName ="SubMenupositive",
+	fakeParam = 0001
+}
+local responseUiParamsValid = {
+	menuID = requestParamsValidData.menuID,
+	menuParams = {
+		position = requestParamsValidData.position,
+		menuName = requestParamsValidData.menuName		
+	}
+}
+local allParamsValid = {
+	requestParamsValidData = requestParamsValidData,
+	responseUiParamsValid = responseUiParamsValid
+}
+
+local requestParamsInvalidData = {
+	menuID = "10",
+	position = "5",
+	menuName = 123,
+	fakeParam = 0001
+}
+
+--[[ Local Functions ]]
+local function addSubMenuValid(pParams)
```

Comment:
@TMyza I propose to remove incoming parameter `pParams` from this function and use local variables `requestParamsValidData` and `responseUiParamsValid` inside it without redundant wrapper table `pParams`

```
local function addSubMenuValid()
	local cid = common.getMobileSession():SendRPC("AddSubMenu", requestParamsValidData)
	common.getHMIConnection():ExpectRequest("UI.AddSubMenu", responseUiParamsValid)
	:Do(function(_, data)
		common.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", {})
	end)
	common.getMobileSession():ExpectResponse(cid, { success = true, resultCode = "SUCCESS" })
	common.getMobileSession():ExpectNotification("OnHashChange")
end
```
The same approach is proposed for function `addSubMenuInvalidData` too.

Proposed approach allows to remove redundant wrapper table `allParamsValid` and improve code readability.